simplifier/visitor/to_normal_form_visitor.py

```diff
@@ -62,14 +63,16 @@ def _multiplying_out_a_term(self, operation: Union[BitwiseAnd, BitwiseOr]):
         For DNF-form, we have an AND-operation and search for an operand that is a disjunction to obtain an equivalent OR-formula.
         """
         if simplified := operation.simplify(keep_form=True):
-            self.visit(simplified)
-            return
+            if not simplified.world.compare(simplified, operation):
+                self.visit(simplified)
+                return
         junction = operation.get_junction()
         if junction:
             new_condition = operation.get_equivalent_condition_of_negated_type(junction)
             if simplified_cond := new_condition.simplify(keep_form=True):
-                self.visit(simplified_cond)
-                return
+                if not simplified_cond.world.compare(simplified_cond, new_condition):
+                    self.visit(simplified_cond)
+                    return
```

Comment:
I think this should work here as well. We have the new condition that would for CNF-form be a BitwiseAnd. 
If we simplify it and it is still a BitwiseAnd, we call want to visit each operand, otherwise, we visit the whole formula.

Reply:
Same as above.